Web.md

```diff
@@ -22,6 +22,82 @@ and retrieve compiled modules from offline storage, instantiate compiled modules
 with JavaScript imports, call the exported functions of instantiated modules,
 alias the exported memory of instantiated modules, etc.
 
+In addition to the core JavaScript API, the Web embedding includes additional
+methods useful in that context.
+Non-web embeddings are not required to support these additional methods.
+
+### Additional Web Embedding API
+
+#### `WebAssembly.compileResponse`
+
+The `compileResponse` function is overloaded based on types of its arguments.
+If none of the following overloads match, then the returned `Promise` is
+[rejected](http://tc39.github.io/ecma262/#sec-rejectpromise)
+with a [`TypeError`](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror).
+
+```
+Promise<WebAssembly.Module> compileResponse(Response source)
+
+Promise<WebAssembly.Module> compileResponse(Promise<Response> source)
+```
+
+The source of the material to compile can be
+a [`Response`](https://fetch.spec.whatwg.org/#response-class)
+or a Promise of a
+[`Response`](https://fetch.spec.whatwg.org/#response-class).
+
+This function starts an asynchronous task to compile a `WebAssembly.Module`
+as described in the [`WebAssembly.Module` constructor](#webassemblymodule-constructor).
+On success, the `Promise` is [fulfilled](http://tc39.github.io/ecma262/#sec-fulfillpromise)
+with the resulting `WebAssembly.Module` object. On failure, the `Promise` is
+[rejected](http://tc39.github.io/ecma262/#sec-rejectpromise) with a
+`WebAssembly.CompileError`.
+
+The `Response` or `Promise<Response>` is used as the source of the
+bytes to compile.
+WebAssembly `source` data is expected to have a mime type of `application/wasm`.
```

Comment:
`application/wasm` sounds reasonable. In the current `<script type="module">` pipeline, we have a restriction that the MIME type of the response should be one of JS MIME type. This is great since we can choose the different way to produce different modules (JS v.s. WASM modules) based on the MIME type of the given Response.